### PR TITLE
docs: add an initial development guide

### PR DESCRIPTION
**Pull request by phlogistonjohn**
OK, I may have been too wordy, but its a start.
Let's treat this as something to build and improve on in the future rather than the end-all-be-all doc. ;-)

### CODE REVIEW

**Comment by phlogistonjohn** [comment]
The plan is to merge this on 2021-06-19 unless major issues are raised. Since this is "only" a doc there's no reason why we can't merge it and then iterate on it afterwards.

**Review by ansiwen — APPROVED** [review]
Sounds like english. Approved.
